src/specialOffer/SpecialOffer.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react'
+
+import { Button, ButtonStatus } from '../button'
+import { ButtonGroup } from '../buttonGroup'
+import { Disclaimer } from '../disclaimer'
+import { Paragraph } from '../paragraph'
+import { SubHeader } from '../subHeader'
+import { StyledSpecialOffer, StyledSpecialOfferPicture } from './SpecialOffer.style'
+
+type Picture = Readonly<{
+  url: string
+  alt: string
+}>
+
+type Link = Readonly<{
```

Comment:
i would merge Link and Button to simplify the code (maybe ActionButton as a tentative name)
```
type ActionButton = {
  label: string
  url?: string
  onClick?: Function
}
```
As well, you might want to have both onClick and href at the same time one day or you might want to mix Link/Button cases. So writing more code to separate them sounds useless and less flexible.
This will divide by 2 the template code to render this and remove the unusual method overload to trigger the correct rendering.
As well, having actions with both optional href/onClick pattern is used in other places in the code.

I would as well remove the constraint on the array size (1 or 2), and just take an array of any size, render them all. That would be the responsibility of the caller to give the right amount; the component can render 0 or 40 ActionButtons.

Reply:
> i would merge Link and Button to simplify the code (maybe ActionButton as a tentative name)
> 
> ```
> type ActionButton = {
>   label: string
>   url?: string
>   onClick?: Function
> }
> ```

I'm not confortable allowing `{ label }` as a valid action. So maybe this type?

```
type ActionButton = {
  label: string
  url: string
} | {
  label: string
  onClick: Function
} | {
  label: string
  url: string
  onClick: Function
}
```

I'm still not confortable with this type to merge both link and button as it's complex and allow something we don't need for now (`url` and `onClick` at the same time) but we will have to maintain. I was surprised we can use a Button as a link in the first place actually.

> As well, you might want to have both onClick and href at the same time one day or you might want to mix Link/Button cases. 

Ok. I'm more into "small API first, open the API after if needed". As both url + onClick combo and mixing Link/Button are not a current need, I see them as *bugs* that Typescript can prevent. 

> I would as well remove the constraint on the array size (1 or 2), and just take an array of any size, render them all. That would be the responsibility of the caller to give the right amount; the component can render 0 or 40 ActionButtons.

Idem, those contraints have been discussed with Heloise and Naima so my position is to embed them in code, for the knowledge to written clearly and not forgotten or bypassed by the caller. The "small API first, open the API after if needed" will allow us to open for more cases if needed for free.

What do you think? Have you a strong opinion for this `ActionButton`?

Reply:
I don't feel strongly about these; let's keep the constraints level high, in particular if you're confident that it should not move too much.

Reply:
+1 for the factorization

I like it this way, simple enough IMHO:
```
type ActionButton = {
  label: string
  url?: string
  onClick?: Function
}
```
Which would give a valid button (with no action, I admit it's strange but at least there is no error/crash)
```
            <Button key={button.label}>
              {button.label}
            </Button>
```

Then if you want to enforce the type with the complete API why not, it's also fine, as you wish 👍

Reply:
I would stay with the link/button distinction as I'm not confortable allowing users of the component to create something broken

Reply:
Since I've proposed in another thread I also vote for it :) hihiihihi

Reply:
> Since I've proposed in another thread I also vote for it :) hihiihihi

Yup. Actually I see two points in this discussion: the renaming of `ctaLinks` and `ctaButtons` to `actions` (done, thanks!) and the merge of the two types (I prefer to not introduce a type that allow something broken). I think we came up with a good tradeoff, thanks everyone!